src/modules/session-recorder/types.ts

```diff
@@ -64,6 +66,7 @@ export type NetworkEventData = CommonEventData & {
 export interface RQSessionEventDataType {
   [RQSessionEventType.RRWEB]: RRWebEventData;
   [RQSessionEventType.NETWORK]: NetworkEventData;
+  [RQSessionEventType.STORAGE]: StorageEventData;
 }
```

Comment:
```suggestion
export interface RQSessionEventDataType {
  [RQSessionEventType.RRWEB]: RRWebEventData;
  [RQSessionEventType.NETWORK]: NetworkEventData;
  [RQSessionEventType.STORAGE]:{
    "local_storage": StorageEventData,
    "session_storage": StorageEventData
  };
}
```

@RuntimeTerror10 WDYT? Can we structure it like this?

Then we can rename `handleStorageEvent` to `addStorageEvent` ->

```js
#addStorageEvent(storageType,event: RQStorageEvent): void {
    const previousSessionStorageEvents = this.#lastTwoSessionEvents[1]?.[STORAGE]?.[storageType];
    previousSessionEvents?.push(event);
  }
```
This method will be independent of the sessionEvent. WDYT?